### PR TITLE
d_a_tbox OK

### PR DESCRIPTION
**Pull request by Sage-of-Mirrors**

### CODE REVIEW

File: src/d/actor/d_a_tbox.cpp [hunk 1]
```diff
@@ -693,53 +694,200 @@ s32 daTbox_c::CreateInit() {
...
+
+    if (mAppearEvtTimer == 0x04 && mSmokeCB.getEmitter() != NULL) {
+        mSmokeCB.end();
+    }
+
+    if (mAppearEvtTimer != 0x00) {
+        mAppearEvtTimer--;
+    }
+
+    if (mpAppearRegAnm->play() != FALSE) {
```
Comment (LagoLunatic) [inline src/d/actor/d_a_tbox.cpp:887]: does `if (mpAppearRegAnm->play()) {` not match?

File: src/d/actor/d_a_tbox.cpp [hunk 2]
```diff
@@ -943,6 +1249,94 @@ s32 daTbox_Draw(daTbox_c* i_tbox) {
...
+
+    if (checkOpen() == FALSE) {
+        dMap_drawPoint(5, current.pos.x, current.pos.y, current.pos.z, mRoomNo, -0x8000, openFlag, mGbaName, 0);
+    }
+
+    mTevStr.mRoomNo = mRoomNo;
+    g_env_light.settingTevStruct(TEV_TYPE_ACTOR, getPositionP(), &mTevStr);
+
+    if (getFuncType() == FUNC_TYPE_TACT) {
+        J3DModelData* platMdlData = mTactPlatformMdl->getModelData();
```
Comment (LagoLunatic) [inline src/d/actor/d_a_tbox.cpp:1273]: might wanna rename to `mpTactPlatformMdl` to make it more clear it's a pointer

**Comment by Sage-of-Mirrors** [comment]
`d_a_tbox` is now OK when it comes to data/rodata/text, but there is a weird bug with the relocation tables that I'm not sure how to fix.

**Comment by Sage-of-Mirrors** [comment]
Relocation bug fixed - had some PTMFs in the wrong order.

Closes #193.

**Pull request merged**